GitCommands/Patches/PatchProcessor.cs

```diff
@@ -15,13 +15,13 @@ private enum PatchProcessorState
             OutsidePatch
         }
 
-        [GeneratedRegex(@"^diff --(?<type>git|cc|combined)\s")]
+        [GeneratedRegex(@"^diff --(?<type>git|cc|combined)\s", RegexOptions.ExplicitCapture)]
         private static partial Regex PatchHeaderRegex();
-        [GeneratedRegex("^diff --git [\\\"]?[abiwco12]/(.*)[\\\"]? [\\\"]?[abiwco12]/(.*)[\\\"]?$")]
+        [GeneratedRegex(@"^diff --git [""]?[abiwco12]/(?<filenamea>.*)[""]? [""]?[abiwco12]/(?<filenameb>.*)[""]?$", RegexOptions.ExplicitCapture)]
```

Comment:
A quote has no special meaning in a regex so I removed the backslash when converting (some strings was changed to verbatim manually, this was in VS and manually changed).
But group is not needed, it can be written without brackets too. I just found that harder to read, with a very low preference and can change

```suggestion
[GeneratedRegex(@"^diff --git ""?[abiwco12]/(?<filenamea>.*)""? ""?[abiwco12]/(?<filenameb>.*)""?$", RegexOptions.ExplicitCapture)]
```

Reply:
I'm a little lost with all these double-quote and escaping making it not so easy to read.
Don't you think the new c#11 raw string literals could be of some help here?
https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/strings/#raw-string-literals

Reply:
A raw string (enclose with """) will not require that the quote is double as in verbatim strings. So a minor benefit.
I want to have a common way to write the regex, to not get separate standards.
To change to verbatim strings would require more modifications, so I avoided that.

Reply:
> A quote has no special meaning in a regex

1. I did not get the intention of the old regex: "What shall `[\"]?` mean? Match escaped quotes? But would not have worked." That's why I thought you had changed the regex.
2. Yes, `@"[""]"` emphasizes that it matches a single character.
3. Raw strings have other drawbacks: Where does the actual string start? How many quotes to use?

Let's keep it as initially proposed.

Reply:
`[\\\"]?` is matching 0 or 1 "
The first `\\` is not needed
It was such details I wanted to address

Will then not change this further, probably merging tomorrow or so